### DISCUSSION TITLE
Mesh data table and correction visualizer messed up

### DISCUSSION THREAD

**Discussion by henryjeynes** [jneilliii/OctoPrint-BedLevelVisualizer#597] [Q&A] [answered]
![Screenshot 2023-02-27 at 17-45-16 Lifesavers-Ender OctoPrint](https://user-images.githubusercontent.com/23265815/221703104-d7ed4a20-7584-4912-8262-97d072f07ad9.png)
![Screenshot 2023-02-27 at 17-46-07 Lifesavers-Ender OctoPrint](https://user-images.githubusercontent.com/23265815/221703889-ab849102-4eff-4d44-81d7-57eca3216a8a.png

I increased the width of .container and .span8 with Themeify, but it's the same way with those disabled.

**Comment by jneilliii** [comment]
I've only seen the data get messed up like this when the probing points were not equal in both x and y direction, so the probing was like a 2x4 grid instead of a 4x4 grid for example. Other than that it typically only happens when the data gets corrupted and re-probing sometimes will work itself out.

**Comment by henryjeynes** [comment]
Indeed the bed topography report is missing some values:
![topography](https://user-images.githubusercontent.com/23265815/221734856-bfbb9deb-c8cd-4d8b-9036-b59116b16520.png)
Probably not related to BedLevelVisualizer. When I figure it out I'll update this.

**Comment by jneilliii** [comment]
`G29 P3` is your friend...

**Reply by jneilliii** [reply]
assuming UBL

**Comment by henryjeynes** [comment, marked as answer]
I had compiled the firmware with the baud rate set to 500k. I think the mainboard (Creality 4.2.7) and/or Octoprint didn't like that for some reason and some characters (esp. newlines) weren't being received properly. Set it back to 250k and everything works correctly.
